src/System.Drawing.Common/src/System/Drawing/Pen.cs

```diff
@@ -669,6 +672,11 @@ public DashStyle DashStyle
                 {
                     EnsureValidDashPattern();
                 }
+
+                if (value != DashStyle.Solid)
+                {
+                    this.dashStyleWasChanged = true;
```

Comment:
Should this be set back to false if the value is Solid? Or rather:

`_dashStyleIsNotSolid = value != DashStyle.Solid;`

Reply:
The `DashStyle_Custom` unit test best explains the behavior:
https://github.com/dotnet/corefx/pull/23797/files#diff-7e1bf1ca3f411d193feac1a3697b2db1R304

* By default, the dash style is solid. If you execute `get_DashPattern`, you'll get an `OutOfMemoryException`. 
* If you change the dash style to a custom dash style, you can get/set `DashPattern` without issue
* If you afterwards change the dash style back to `Solid`,  calling `get_DashPattern` will _not_ cause an `OutOfMemoryException`. 

So the flag really tracks whether the dash patter is or has been different from Solid at any point or time. `dashStyleIsNotSolid` doesn't feel right because it's present tense; `dashStyleIsOrWasNotSolid` is the best I can come up with, let me know if you have a better idea for the name :).

Reply:
Okay, sounds like a reasonable description of what the field is tracking. Very weird behavior from GDI+ there...